momo_modules/multi_sensor_module/src/app_main.c

```diff
@@ -174,17 +195,14 @@ void acquire_pulse()
 
 void read_pulses()
 {
-	mib_buffer[0] = pulse_count() & 0xFF;
-	mib_buffer[1] = pulse_count() >> 8;
+	mib_buffer[0] = counter & 0xFF;
```

Comment:
Won't this cause unpredictable values if called after a few periods have past?  For instance, if the board is acquiring pulses every 4 seconds and reporting them every minute, then somebody tries to inspect the pulse count after 30 seconds, the caller will get 30 seconds worth of pulses (with no indication of the period) and then the counter will be reset.

An alternative:
return both the `counter` and `periods` values, don't reset the counter, and compute the pulse/second value dynamically on the receiving (python) side.